packages/issuer/src/blockchain-facade/Certificate.ts

```diff
@@ -0,0 +1,257 @@
+import { TransactionReceipt, EventLog } from 'web3-core';
+
+import { Configuration, BlockchainDataModelEntity, Timestamp } from '@energyweb/utils-general';
+
+import { Registry, PublicIssuer, PUBLIC_CERTIFICATE_TOPIC } from '..';
+
+export interface ICertificate {
+    id: string;
+
+    issuer: string;
+    
+    deviceId: string;
+    generationStartTime: number;
+    generationEndTime: number;
+
+    energy: number;
+    creationTime: number;
+
+    sync(): Promise<ICertificate>;
+
+    getCertificateOwner(): Promise<string>;
+    isClaimed(): Promise<boolean>;
+    claim(amount: number): Promise<TransactionReceipt>;
+
+    transfer(to: string, amount: number): Promise<TransactionReceipt>;
+
+    getAllCertificateEvents(): Promise<EventLog[]>;
+}
+
+const getAccountFromConfiguration = (configuration: Configuration.Entity) => ({
+    from: configuration.blockchainProperties.activeUser.address,
+    privateKey: configuration.blockchainProperties.activeUser.privateKey
+});
+
+export const getAllCertificateEvents = async (
+    certId: number,
+    configuration: Configuration.Entity
+): Promise<EventLog[]> => {
+    const registry: Registry = configuration.blockchainProperties.certificateLogicInstance;
+
+    const allEvents = await registry.getAllEvents(
+        {
+            topics: [
+                null,
+                configuration.blockchainProperties.web3.utils.padLeft(
+                    configuration.blockchainProperties.web3.utils.fromDecimal(certId),
+                    64,
+                    '0'
+                )
+            ],
+            toBlock: undefined
```

Comment:
Please make sure `fromBlock` is passed (even if it's 0, so it works on Volta).

Also, maybe instead of `toBlock` undefined it's better to say `toBlock: 'latest'`?

Reply:
This was copied from old Certificate.ts, will update.